src/main/java/org/opensearch/neuralsearch/settings/NeuralSearchSettings.java

```diff
@@ -46,4 +55,40 @@ public final class NeuralSearchSettings {
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
+
+    public static Setting<Integer> SPARSE_ALGO_PARAM_INDEX_THREAD_QTY_SETTING = Setting.intSetting(
+        SPARSE_ALGO_PARAM_INDEX_THREAD_QTY,
+        SPARSE_DEFAULT_ALGO_PARAM_INDEX_THREAD_QTY,
+        -1, // -1 means that user did not give specific thread quantity
+        INDEX_THREAD_QTY_MAX,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    public static FixedExecutorBuilder updateThreadQtySettings(Settings settings) {
```

Comment:
nit: rename to getExecutorBuilder?

Reply:
What about one function in NeuralSearchSettings.java to update setting, one function in ClusterTrainingRunning to getExecutorBuilder?